core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserUniqueIDDomainResolver.java

```diff
@@ -233,4 +240,37 @@ private Connection getDBConnection() throws SQLException, UserStoreException {
         }
         return dataSource.getConnection();
     }
+
+    /**
+     * Commit the transaction.
+     *
+     * @param dbConnection database connection.
+     */
+    private void commitTransaction(Connection dbConnection) {
+
+        try {
+            if (dbConnection != null && !dbConnection.getAutoCommit()) {
+                dbConnection.commit();
+            }
+        } catch (SQLException e) {
+            rollbackTransaction(dbConnection);
+            log.error("An error occurred while commit transactions. ", e);
+        }
+    }
+
+    /**
+     * Revoke the transaction when catch then sql transaction errors.
+     *
+     * @param dbConnection database connection.
+     */
+    private void rollbackTransaction(Connection dbConnection) {
+
+        try {
+            if (dbConnection != null) {
```

Comment:
Should check whether autocommit is false here

Reply:
done